packages/material-ui/src/DialogTitle/DialogTitle.js

```diff
@@ -1,31 +1,66 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
-import withStyles from '../styles/withStyles';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
 import Typography from '../Typography';
+import experimentalStyled from '../styles/experimentalStyled';
+import useThemeProps from '../styles/useThemeProps';
+import { getDialogTitleUtilityClass } from './dialogTitleClasses';
 
-export const styles = {
-  /* Styles applied to the root element. */
-  root: {
+const overridesResolver = (props, styles) => styles.root || {};
+
+const useUtilityClasses = (styleProps) => {
+  const { classes } = styleProps;
+
+  const slots = {
+    root: ['root'],
+  };
+
+  return composeClasses(slots, getDialogTitleUtilityClass, classes);
+};
+
+const DialogTitleRoot = experimentalStyled(
+  'div',
+  {},
+  {
+    name: 'MuiDialogTitle',
+    slot: 'Root',
+    overridesResolver,
+  },
+)(() => {
+  return {
+    /* Styles applied to the root element. */
     margin: 0,
     padding: '16px 24px',
     flex: '0 0 auto',
-  },
-};
+  };
+});
 
-const DialogTitle = React.forwardRef(function DialogTitle(props, ref) {
-  const { children, classes, className, disableTypography = false, ...other } = props;
+const DialogTitle = React.forwardRef(function DialogTitle(inProps, ref) {
+  const props = useThemeProps({
+    props: inProps,
+    name: 'MuiDialogTitle',
+  });
+
+  const { children, className, disableTypography = false, ...other } = props;
+  const styleProps = { ...props };
```

Comment:
```suggestion
  const styleProps = { ...props, disableTypography };
```

Reply:
@mnajdova Should disableTypogtaphy be passed to stypesProps equal to what is only used as a switch to toggle a component?